src/match/controllers/create_match_controller.py

```diff
@@ -0,0 +1,46 @@
+from api.utils import failure_response
+from api.utils import success_response
+from django.contrib.auth.models import User
+from match import match_status
+from match.models import Match
+from rest_framework import status
+
+
+class CreateMatchController:
+    def __init__(self, data, serializer):
+        self._data = data
+        self._serializer = serializer
+
+    def process(self):
+        """Process a request to create a match."""
+        match_ids = self._data.get("ids")
+        if match_ids is None:
+            return failure_response(
+                "POST body is misformatted", status.HTTP_400_BAD_REQUEST
+            )
+        user_1 = User.objects.filter(id=match_ids[0])
+        user_2 = User.objects.filter(id=match_ids[1])
+        if not user_1:
```

Comment:
Check if not `user_1` before filtering for `user_2` so that we immediately send a failure response if `user_1` is None and don't waste computational power on the second unnecessary filter.